### ISSUE TITLE
Program launch issue

### ISSUE THREAD

**Issue by Mmoi-Fr** [sandboxie-plus/Sandboxie#3173]
### Describe what you noticed and did

When I set the ConfidentialBox=y option on a box (which already was on in previous versions), the forced flder's programs are started in a strange state.
They are listed on my Windows user and kind of frozen.
I must right click on the sandboxie panel, on the program and then click on resume to make it run.
Speaking of Firefox, all subprocesses are sandboxed (ANONYMOUS LOGON) but not the "main one".

Moreover, I don't know if something changed in the newer versions (I was on 1.9.2) but the username listed on all task managers now are "ANONYMOUS LOGON" and no more the box name.

I hope the problem is clear enough.
Let me know if you want me to install inbetween version to check when the regresion was introduced.

### How often did you encounter it so far?

At every launch

### Affected program

Firefox 115.0.0.3 (any version should be affected), jdownloader...

### Download link

Not relevant

### Where is the program located?

The program is installed only outside the sandbox.

### Expected behavior

Firefox is launched with the "Firefox" (box name) (or the ANONYMOUS LOGON) user / token and starts not frozen

### What is your Windows edition and version?

Windows 10 LTSC 21H2

### In which Windows account you have this problem?

A local account (Administrator).

### Please mention any installed security software

None

### What version of Sandboxie are you running?

Sandboxie Plus 10.0.3

### Is it a new installation of Sandboxie?

I recently did a new clean installation.

### Is it a regression?

Worked well on 1.9.2 (was broken on 1.9.6 if I remeber correctly)

### In which sandbox type you have this problem?

In a hardened sandbox with data protection (red sandbox icon).

### Can you reproduce this problem on a new empty sandbox?

Not relevant to my request.

### Did you previously enable some security policy settings outside Sandboxie?

No

### Crash dump

No crash

### Trace log

No logs seems to be generated

### Sandboxie.ini configuration

```shell
[GlobalSettings]
DefaultBox=DefaultBox
FileRootPath=\??\%SystemDrive%\Sandbox\%USER%\%SANDBOX%
KeyRootPath=\REGISTRY\USER\Sandbox_%USER%_%SANDBOX%
IpcRootPath=\Sandbox\%USER%\%SANDBOX%\Session_%SESSION%
NetworkEnableWFP=y
EditAdminOnly=y
Template=WindowsRasMan
Template=OfficeLicensing
Template=OfficeClickToRun
Template=RTSS
TemplateReject=WindowsLive
ForceDisableSeconds=60
MonitorStackTrace=y


[Jdownloader]

Enabled=y
AutoRecover=n
BlockNetworkFiles=y
BorderColor=#00ff72,off,2
Template=FileCopy
Template=BlockPorts
Template=LingerPrograms
ConfigLevel=10
UseSecurityMode=y
UsePrivacyMode=y
DropAdminRights=y
AllowNetworkAccess=<BlockNetAccess>,n
BoxNameTitle=n
ShowForRunIn=n
ClosePrintSpooler=y
CopyLimitKb=-1
ConfidentialBox=y
AutoDelete=y
ClosedFilePath=<BlockNetDevices>,InternetAccessDevices
ForceFolder=C:\Program Files\JDownloader 2.0
OpenPipePath=C:\Program Files\JDownloader 2.0
```

**Comment by isaak654** [comment]
> Moreover, I don't know if something changed in the newer versions (I was on 1.9.2) but the username listed on all task managers now are "ANONYMOUS LOGON" and no more the box name.

In this case you're probably looking for [`SandboxieLogon=y`](https://github.com/sandboxie-plus/Sandboxie/discussions/2064#discussioncomment-3269038), which is no longer a default setting since [v1.2.7](https://github.com/sandboxie-plus/Sandboxie/releases/tag/v1.2.7).

**Comment by bastik-1001** [comment]
When I add `ConfidentialBox=y` to a sandbox, which has a forced folder rule and a forced program (not to say this is conditional), the program gets launched under Sandboxie's control, but it does not show up. Sandboxie Plus, shows the process as being suspended and when I click resume, the window appears.

**Comment by offhub** [comment]
@bastik-1001 Try this:
```
DenyHostAccess=explorer.exe,n
```

**Comment by bastik-1001** [comment]
@offhub Thank you. This works.

Will that be the solution, or is this a workaround?

**Comment by offhub** [comment]
Let's just consider this a workaround until @DavidXanatos says otherwise. 😁 You should also apply similar settings for other file managers, if you use them.

**Comment by bastik-1001** [comment]
I did not consider (nor did I test) other file managers. Thank you for bringing that to everyone's attention. It's true that David will need to decide if that is doing a good enough job, or if that should work without exemptions.

If this workaround is added into a release to make default Windows explorer work with forced settings and the confidential setting, by default, without intending to fix it, the documentation should be updated to include your point with file managers needing to be exempt for it to work.

**Comment by Mmoi-Fr** [comment]
> @bastik-1001 Try this:
> 
> ```
> DenyHostAccess=explorer.exe,n
> ```

This (replacing with firefox.exe, obviously) did not work for me as firefox still was launched paused, without the sandbox user. (tested with Sandboxie-Plus-x64-v1.11.0)

**Comment by bastik-1001** [comment]
@Mmoi-Fr 

Can you please try with explorer.exe? (or the file manager you use to start the program, like Firefox, with.) For me, it was not the program that got launched, but the program that launched the binary.

**Comment by Mmoi-Fr** [comment]
> @Mmoi-Fr
> 
> Can you please try with explorer.exe? (or the file manager you use to start the program, like Firefox, with.) For me, it was not the program that got launched, but the program that launched the binary.

Oh, I didn't catch that in this way, I just tried and it seems to work, thanks
I'll let the tcket open in order to keep track of the issue, until a fix is (or isn't) released

**Comment by offhub** [comment]
`DenyHostAccess=audiodg.exe,n`

**Comment by DavidXanatos** [comment]
there will be a fix in the next build to allows one time access to the process when creating it, which will be more secure.
also I'll add a template containing 
DenyHostAccess=audiodg.exe,n

to which we can add more usefull excemptions

**Comment by bjm234** [comment]
> In this case you're probably looking for [`SandboxieLogon=y`](https://github.com/sandboxie-plus/Sandboxie/discussions/2064#discussioncomment-3269038), which is no longer a default setting since [v1.2.7](https://github.com/sandboxie-plus/Sandboxie/releases/tag/v1.2.7).

Hmm, interesting my Global remains SandboxieLogon=y

**Comment by isaak654** [comment]
> Hmm, interesting my [Global Settings] remains SandboxieLogon=y

For those who set it up before [v1.2.7](https://github.com/sandboxie-plus/Sandboxie/releases/tag/v1.2.7), nothing is changed.
I personally have always been against making that security option as non-default, so I would suggest to any Sandboxie user who cares about security to keep it enabled (at least for a while) and report any third-party incompatibility to the respective vendor.

**Comment by isaak654** [comment]
> Hmm, interesting my [Global Settings] remains SandboxieLogon=y

For those who set it up before [v1.2.7](https://github.com/sandboxie-plus/Sandboxie/releases/tag/v1.2.7), nothing is changed.
I personally have always been against making that security option as non-default, so I would suggest to any Sandboxie user who cares about cross-sandbox attacks prevention to keep the option enabled (at least for a while) and report any third-party incompatibility to the respective vendor. If you ask why this was not prioritised, I would like to know too.

**Comment by isaak654** [comment]
> Hmm, interesting my [Global Settings] remains SandboxieLogon=y

For those who set it up before [v1.2.7](https://github.com/sandboxie-plus/Sandboxie/releases/tag/v1.2.7), nothing is changed.
I personally have always been against making that security option as non-default, so I would suggest to any Sandboxie user who cares about cross-sandbox attacks prevention to keep the option enabled (at least for a while) and report any third-party incompatibility to the respective vendor. If you ask why this was not prioritized, I would like to know too.

**Comment by DavidXanatos** [comment]
fixed in 1.12.3
